gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -1056,7 +1056,6 @@ public long nextLong() throws IOException {
    * consumes the closing quote, but does not include it in the returned string.
    *
    * @param quote either ' or ".
-   * @throws NumberFormatException if any unicode escape sequences are malformed.
```

Comment:
Perhaps mention `MalformedJsonException` instead? Though for a private method it's probably not worth it. On the other hand, I think we could change the `throws` clause to `MalformedJsonException`?

Reply:
`MalformedJsonException` is a subclass of `IOException`, but this method (like most of the other `JsonReader` methods) can throw both:
- `MalformedJsonException` for invalid JSON
- `IOException` when reading from the underlying `Reader` fails

I think I am not going to add `throws MalformedJsonException` to the method signature because then for consistency (but not because the compiler requires it) it would have to be done for all other methods as well, and that is probably not worth it.

However, I have adjusted two internal methods now which always throw `MalformedJsonException` (`checkLenient` and `syntaxError`).

Reply:
OK, thanks!